deploy/plays/copy.yml

```diff
@@ -0,0 +1,100 @@
+- hosts: machines
+  vars:
+    main_server_public: '{{ install_dir }}/main_server/public'
+  tasks:
+
+    - name: Create the installation directory
+      file:
+        path: '{{ install_dir }}'
+        state: directory
+        owner: {{ ansible_user }}
+        recurse: yes
+      become: yes
+
+    - name: Create the subdirectories under installation directory
+      file:
+        path: '{{ install_dir }}/tmp'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/gitlab'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/gitlab/config'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/gitlab/data'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/log/gitlab'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/mysql'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/log/mysql'
+        state: directory
+
+    - name: Copy files to main_server subdirectory of installation directory
+      file:
+        path: '{{ install_dir }}/main_server'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/log/main_server'
+        state: directory
+    - copy:
+        src: 'main_server/'
+        dst: '{{ install_dir }}/main_server/'
+      args:
+        chdir: '{{ playbook_dir }}/../'
+
+    - name: Copy files to load_balancer subdirectory of installation directory
+      file:
+        path: '{{ install_dir }}/load_balancer'
+        state: directory
+    - file:
+        path: '{{ install_dir }}/log/load_balancer'
+        state: directory
+    - copy:
+        src: 'load_balancer/'
+        dst: '{{ install_dir }}/load_balancer/'
+      args:
+        chdir: '{{ playbook_dir }}/../'
+
+    - name: Creating directories for execution nodes
```

Comment:
As per Ansible docs of [copy module](http://docs.ansible.com/ansible/latest/modules/copy_module.html), the destination directory is created if it does not exist. Please see the explanation under `dest` property of the page. Hence, we don't need to create the following directories:
```
{{ install_dir }}/execution_nodes/execution_node_{{ item }}
```